### PR TITLE
[SEO] Add meta tags

### PR DESCRIPTION
**Pull request by Tijani-Dia**
This PR adds some meta tags for SEO purposes.

### CODE REVIEW

File: components/Layout/types.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 type LayoutProps = {
     title: string;
     lastPublishedAt: string;
+    meta: { [key: string]: string | boolean | null };
```
Comment (jhancock532) [inline components/Layout/types.ts:4]: **Nitpick**: This is a neat type definition shorthand, but as we don't need to support completely custom keys in our meta object (just `socialText`, `searchDescription`, `socialImageUrl`, `seoTitle`, `slug`) I'd prefer if these fields were specifically defined in the `meta` props type definition. This way you also don't have to recast the object field types every time you use them in `Layout.tsx`

File: components/Layout/Layout.tsx [hunk 2]
```diff
@@ -6,11 +6,75 @@ import FooterSlackCTA from './FooterSlackCTA';
...
+                property="og:url"
+                content={`https://areweheadlessyet.wagtail.org/${
+                    meta.slug !== 'are-we-headless-yet' ? meta.slug : ''
+                }`}
+            />
+            <meta
+                property="og:title"
+                content={meta.seoTitle ? (meta.seoTitle as string) : title}
+            />
+            <meta property="og:image" content={meta.socialImageUrl as string} />
```
Comment (jhancock532) [inline components/Layout/Layout.tsx:33]: Please could we also add support for Twitter here: open graph covers pretty much everything else 👍

```
<meta property="twitter:card" content="summary_large_image">
<meta property="twitter:url" content="...">
<meta property="twitter:title" content="...">
<meta property="twitter:description" content="...">
<meta property="twitter:image" content="...">

```

**Comment by jhancock532** [comment]
Looks great, two minor fixes then I'm happy for this to be merged 🚀

File: components/Layout/types.ts (hunk 1 above)
Comment (Tijani-Dia) [reply components/Layout/types.ts]: Nice suggestion. I've updated it. Thanks!

File: components/Layout/Layout.tsx (hunk 2 above)
Comment (Tijani-Dia) [reply components/Layout/Layout.tsx]: Added.

**Comment by Tijani-Dia** [comment]
Thanks, @jhancock532. I've addressed your comments. Let me know if you're happy with the changes.

**Pull request merged**